source/main.cpp

```diff
@@ -208,8 +210,42 @@ int main(int argc, char *argv[])
 		if(isTesting && !noTestMute)
 			Audio::SetVolume(0);
 
-		// This is the main loop where all the action begins.
-		GameLoop(player, conversation, testToRunName, debugMode);
+		// In case the player wants to reload.
+		future<void> dataReloading;
+
+		// This loop is for the case of a data reload.
+		do {
+			Messenger::SetReload(false);
```

Comment:
This is not really a clean way of solving the problem. Probably best to do it using the existing Panel infrastructure instead of drawing this manually here in main. So say like clearing all panels, adding a new panel (kinda similar to GameLoadingPanel) that reloads all the resources.

Maybe you can even reuse GameLoadingPanel by doing some refactoring?  Say by moving some common initializing code there instead of doing it in main. I'm not sure though.